src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,182 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            this.add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> x = node(index);
```

Comment:
```suggestion
        Node<T> node = getNode(index);
```
x name is a bad practice, choose full words